utils/connection_handler.py

```
@@ -0,0 +1,107 @@
+            initial_backoff: Initial wait time in seconds
+            backoff_factor: Multiplier for exponential backoff
+        """
+        self.max_retries = max_retries
+        self.initial_backoff = initial_backoff
+        self.backoff_factor = backoff_factor
+        self.failed_attempts = 0
+        self.successful_retries = 0
+        self.total_calls = 0
+        logger.info(f"🔌 Connection handler initialized: max_retries={max_retries}, initial_backoff={initial_backoff}s")
+    
+    def retry_with_backoff(self, func: Callable, *args, **kwargs) -> Optional[Any]:
```

Ruff (0.14.10)

Missing type annotation for `*args`

(ANN002)